### PR TITLE
Add more characters to the On-Screen-Keyboard

### PR DESCRIPTION
**Pull request by Megamouse**
1. Fixes an OSK regression that exchanged 'l' and 'L' with 'U' due to a find/replace. 🤦‍♂ 
2. Moves some OSK code into more specific files
3. Adds more characters to the english osk key layout in form of a mode switcher for: normal charset, extended charset, special characters

The charset was chosen based on official documentation for the english PS3 OSK.

### CODE REVIEW

File: rpcs3/Emu/RSX/Overlays/overlay_osk.cpp [hunk 1]
```diff
@@ -647,7 +647,7 @@ namespace rsx
 				{{U"h", U"H"}, default_bg, 1},
 				{{U"j", U"J"}, default_bg, 1},
 				{{U"k", U"K"}, default_bg, 1},
-				{{U"U", U"U"}, default_bg, 1},
+				{{U"l", U"L"}, default_bg, 1},
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Overlays/overlay_osk.cpp]: 😆

File: rpcs3/Emu/RSX/Overlays/overlay_osk.cpp [hunk 2]
```diff
@@ -254,16 +275,25 @@ namespace rsx
...
 				const u32 current_index = (selected_y * num_columns) + selected_x;
-				const u32 output_count = ::size32(m_grid[current_index].outputs);
+				const auto current_cell = m_grid[current_index];
+
+				std::vector<std::u32string> outputs;
+
+				if (m_selected_mode < layer_mode::mode_count && m_selected_mode < current_cell.outputs.size())
+				{
+					outputs = current_cell.outputs[m_selected_mode];
+				}
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Overlays/overlay_osk.cpp]: This block and a similar one elsewhere are copying an array of strings every time a button is pressed which seems pointless and wasteful. You could literally flag a bool here to true and use that for the condition below.

File: rpcs3/Emu/RSX/Overlays/overlay_osk.cpp [hunk 3]
```diff
@@ -543,7 +590,14 @@ namespace rsx
 					{
 						w--;
 
-						if (!c.outputs.empty())
+						std::vector<std::u32string> outputs;
+
+						if (m_selected_mode < layer_mode::mode_count < c.outputs.size())
+						{
+							outputs = c.outputs[m_selected_mode];
+						}
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Overlays/overlay_osk.cpp]: See comment for similar block.

Comment (Megamouse) [reply rpcs3/Emu/RSX/Overlays/overlay_osk.cpp]: Fixed I guess.
Also fixed the nasty typo in the if condition

File: rpcs3/Emu/RSX/Overlays/overlay_osk.cpp (hunk 2 above)
Comment (Megamouse) [reply rpcs3/Emu/RSX/Overlays/overlay_osk.cpp]: Fixed by using output_count

File: rpcs3/Emu/RSX/Overlays/overlay_osk.cpp [hunk 4]
```diff
@@ -254,16 +275,25 @@ namespace rsx
 			auto on_accept = [&]()
 			{
 				const u32 current_index = (selected_y * num_columns) + selected_x;
-				const u32 output_count = ::size32(m_grid[current_index].outputs);
+				const auto current_cell = m_grid[current_index];
```
[the remark this answers is not in the document]
Comment (Megamouse) [reply rpcs3/Emu/RSX/Overlays/overlay_osk.cpp]: done

**Pull request merged**